src/main/java/net/sf/jabref/gui/openoffice/OpenOfficePanel.java

```diff
@@ -322,6 +332,39 @@ public void actionPerformed(ActionEvent e) {
 
     }
 
+    private void exportEntries() {
+        try {
+            if (style == null) {
+                style = loader.getUsedStyle();
+            } else {
+                style.ensureUpToDate();
+            }
+
+            ooBase.updateSortedReferenceMarks();
+
+            List<BibDatabase> databases = getBaseList();
+            List<String> unresolvedKeys = ooBase.refreshCiteMarkers(databases, style);
+            BibDatabase newDatabase = ooBase.generateDatabase(databases);
+            if (!unresolvedKeys.isEmpty()) {
+                JOptionPane.showMessageDialog(frame,
+                        Localization.lang(
+                                "Your OpenOffice/LibreOffice document references the BibTeX key '%0', which could not be found in your current database.",
+                                unresolvedKeys.get(0)),
+                        Localization.lang("Unable to generate new database"), JOptionPane.ERROR_MESSAGE);
+            }
+
+            Defaults defaults = new Defaults(
+                    BibDatabaseMode.fromPreference(Globals.prefs.getBoolean(JabRefPreferences.BIBLATEX_DEFAULT_MODE)));
+
+            BibDatabaseContext databaseContext = new BibDatabaseContext(newDatabase, defaults);
+            this.frame.addTab(databaseContext, true);
+
+        } catch (Exception e) {
```

Comment:
Well, there are quite a few Exception types already defined in OO/LO so defining one more (in case we want JabRefException to be used later for something else) shouldn't be a problem. It may be a problem naming it since I'm not really sure what the errors are, maybe `ThingsWentWrongException`? ;-)